sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/hash.scala

```diff
@@ -111,8 +112,10 @@ case class Sha2(left: Expression, right: Expression)
         // DigestUtils doesn't support SHA-224 now
         try {
           val md = MessageDigest.getInstance("SHA-224")
-          md.update(input)
-          UTF8String.fromBytes(md.digest())
+          val messageDigest = md.digest(input);
```

Comment:
Actually, hm, now that I look at it - looks like commons-codec (now?) supports SHA-224 too:
https://commons.apache.org/proper/commons-codec/apidocs/org/apache/commons/codec/digest/MessageDigestAlgorithms.html

I wonder if we can just use that now, but also, if its implementation has the same issue? your test would probably help tell.

Reply:
Yep, thanks for catching that! I switched it over to use `DigestUtils` and it seems to work.